### PR TITLE
[9.0] [ADD] project_issue_timesheet_time_control

### PR DESCRIPTION
**Pull request by pedrobaeza**
It depends on https://github.com/OCA/project/pull/163 and includes its commits. Relevant commit is https://github.com/OCA/project/commit/c5375c258a0c14c18e87c0e6647e972e7cf47498
# 
# Project Issue Timesheet Time Control

This module adds a button at project issue timesheets to compute minutes
lasted from start date to the current moment.
# Installation

This module is auto-installed when you install
_project_timesheet_time_control_ and _project_issue_sheet_.

### CODE REVIEW

File: project_issue_timesheet_time_control/README.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+
+====================================
+Project Issue Timesheet Time Control
+====================================
+
+This module adds a button at project issue timesheets to compute minutes
```
Comment (elicoidal) [inline project_issue_timesheet_time_control/README.rst]: ...to compute the spent time, in minutes, from...

File: project_timesheet_time_control/README.rst [hunk 2]
```diff
@@ -0,0 +1,58 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+
+==============================
+Project Timesheet Time Control
+==============================
+
+This module adds a button at account analytic line level to compute minutes
```
Comment (elicoidal) [inline project_timesheet_time_control/README.rst]: see above

File: project_timesheet_time_control/README.rst [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+lasted from start date to the current moment.
+
+It adds a datetime field that replaces ``date`` field in tree view, and write
```
Comment (elicoidal) [inline project_timesheet_time_control/README.rst]: create a list

File: project_timesheet_time_control/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# © 2016 Antonio Espinosa
```
Comment (elicoidal) [inline project_timesheet_time_control/__openerp__.py]: usually companies are copyright holders

Comment (pedrobaeza) [inline project_timesheet_time_control/__openerp__.py]: I don't agree. The copyright is for coders, not companies.

File: project_timesheet_time_control/README.rst (hunk 2 above)
Comment (pedrobaeza) [inline project_timesheet_time_control/README.rst]: These comments goes for the PR https://github.com/OCA/project/pull/163. I have fixed in that commit (and they will be propagated to this PR). Please read the PR main text, where I explained that there's only one relevant commit for this module.

File: project_timesheet_time_control/__openerp__.py (hunk 4 above)
Comment (elicoidal) [inline project_timesheet_time_control/__openerp__.py]: OK

**Comment by pedrobaeza** [comment]
cc @Tecnativa

**Comment by rafaelbn** [comment]
Let's see what's going on with https://github.com/OCA/project/pull/163.

File: project_timesheet_time_control/__openerp__.py (hunk 4 above)
Comment (dreispt) [inline project_timesheet_time_control/__openerp__.py]: @pedrobaeza If you work full time for a company, the company is the copyright owner of your code.

Comment (pedrobaeza) [inline project_timesheet_time_control/__openerp__.py]: No, that depends on the work contract itself. At Tecnativa, we don't remove the copyright from the coder.

File: project_issue_timesheet_time_control/README.rst [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/project/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/OCA/
+project/issues/new?body=module:%20
+project_issue_sheet_time_control%0Aversion:%20
+9.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (yajo) [inline project_issue_timesheet_time_control/README.rst]: Drop this

File: project_issue_timesheet_time_control/views/project_issue_view.xml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+                    <attribute name="invisible">1</attribute>
+                </xpath>
+                <xpath expr="//field[@name='timesheet_ids']//field[@name='date']" position="after">
+                    <field name="date_time"/>
+                </xpath>
+                <xpath expr="//field[@name='timesheet_ids']//field[@name='unit_amount']" position="after">
+                    <button name="button_end_work"
+                            string="Stop"
+                            type="object"
+                            icon="STOCK_STOP"
```
Comment (yajo) [inline project_issue_timesheet_time_control/views/project_issue_view.xml]: See https://github.com/OCA/project/pull/163#discussion_r76428650

File: project_issue_timesheet_time_control/__openerp__.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+# © 2016 Pedro M. Baeza
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+{
+    'name': 'Project issue timesheet time control',
+    'version': '9.0.1.0.0',
+    'category': 'Project Management',
+    'author': 'Serv. Tecnol. Avanzados - Pedro M. Baeza, '
+              'Odoo Community Association (OCA)',
+    'website': 'http://www.serviciosbaeza.com',
```
Comment (yajo) [inline project_issue_timesheet_time_control/__openerp__.py]: Tecnativa?

File: project_issue_timesheet_time_control/views/project_issue_view.xml [hunk 8]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+    <data>
```
Comment (yajo) [inline project_issue_timesheet_time_control/views/project_issue_view.xml]: `<odoo>`

**Comment by yajo** [comment]
Little remarks, :+1: for the rest (once #163 gets in).

**Comment by pedrobaeza** [comment]
@chienandalu @luismontalba @dreispt  you can review this one

**Comment by luismontalba** [comment]
I find a couple of issues:
1)
I go to Timesheets > Time Tracking > Activities.
I create a new record.
I don't select any "Analytic account/project" but I select  an issue:

seleccion_001

The "Analytic account/project" field is not auto-filled and it's not possible to save the record.

2)

I go to Project > Search > Issues.
I create a new one.
On the "Timesheets" page I create a line:

seleccion_002

When I save changes it disappears:

seleccion_003

**Comment by pedrobaeza** [comment]
@luismontalba The problem in both cases is the same: you don't have the analytic account filled. Please see the field with that name in the screenshots...

P.S.: Don't use demo data. Create a new issue as stated in the README.

**Review by luismontalba — APPROVED** [review]
That's right, avoiding data demo it works on runbot.

**Review by rafaelbn — APPROVED** [review]
Tested in runbot

File: project_issue_timesheet_time_control/views/project_issue_view.xml [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+                <attribute name="invisible">1</attribute>
+            </xpath>
+            <xpath expr="//field[@name='timesheet_ids']//field[@name='date']" position="after">
+                <field name="date_time"/>
+            </xpath>
+            <xpath expr="//field[@name='timesheet_ids']//field[@name='unit_amount']" position="after">
+                <button name="button_end_work"
+                        string="Stop"
+                        type="object"
+                        icon="fa-times-circle"
```
Comment (carlosdauden) [inline project_issue_timesheet_time_control/views/project_issue_view.xml:18]: Not important, but usually the symbol to stop something in progress is fa-stop-circle.
Is there any reason to use the blade? Is used usually in odoo?

**Review by carlosdauden — APPROVED** [review]

File: project_issue_timesheet_time_control/views/project_issue_view.xml (hunk 9 above)
Comment (pedrobaeza) [reply project_issue_timesheet_time_control/views/project_issue_view.xml:18]: Not really. It's only mimic of the previous GTK icon we were using. Do you prefer this one? We can certainly switch both.

Comment (carlosdauden) [reply project_issue_timesheet_time_control/views/project_issue_view.xml:18]: What you like, I have already approved the changes :smile:

Comment (pedrobaeza) [reply project_issue_timesheet_time_control/views/project_issue_view.xml:18]: OK, icon changed

Comment (pedrobaeza) [reply project_issue_timesheet_time_control/views/project_issue_view.xml:18]: Unfortunately, that icon is not yet available in the Font-Awesome version included in version 9, so I had to revert back to the cross icon. Let's see for version 10.

**Review by yajo — APPROVED** [review]

**Comment by yajo** [comment]
Come on, merge! :shipit:

**Comment by pedrobaeza** [comment]
Wait a bit to see if the offset of the issue field is happening also here.

**Comment by pedrobaeza** [comment]
OK, the problem was other. This is running perfectly, but @dreispt is always very picky about self-merging, so I let him to merge it.

**Review by chienandalu — APPROVED** [review]
Functionality tested on runbot

**Comment by pedrobaeza** [comment]
@dreispt can you merge?

**Comment by dreispt** [comment]
Done!

**Comment by pedrobaeza** [comment]
Thanks!

**Pull request merged**